pilot/proxy/envoy/config.go

```diff
@@ -707,13 +716,9 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 	// to the service address
 	// assumes that endpoint addresses/ports are unique in the instance set
 	// TODO: validate that duplicated endpoints for services can be handled (e.g. above assumption)
-	for _, instance := range instances {
-		endpoint := instance.Endpoint
-		servicePort := endpoint.ServicePort
-		protocol := servicePort.Protocol
-		cluster := buildInboundCluster(endpoint.Port, protocol, mesh.ConnectTimeout)
-		clusters = append(clusters, cluster)
-
+	buildListerner := func(cluster *Cluster, instanceServiceHostname string,
```

Comment:
nit: we don't follow this inline function thingy in the pilot code base. Can you please create appropriate function and call it explicitly?
also, typo. buildListener